### PR TITLE
Raii locking of the document

### PR DESCRIPTION
**Pull request by Febbe**
Also removed a memory leak in CustomExportJob.cpp

### CODE REVIEW

**Comment by Technius** [comment]
A merge conflict needs to be resolved before this can be reviewed/merged.

File: src/model/Document.cpp [hunk 1]
```diff
@@ -82,7 +83,7 @@ void Document::unlock()
 	//	fprintf(stderr, "\n\n\n\n");
 }
 
-bool Document::tryLock()
+bool Document::try_lock()
```
[the remark this answers is not in the document]
Comment (Febbe) [reply src/model/Document.cpp]: Without the renaming the use of lock guards would not be possible.

**Comment by Febbe** [comment]
I think it would be even better, to replace the GMutex of the Document with an std::shared_mutex and guard reading operations with it. This could increase both safety and responsiveness.

**Comment by Febbe** [comment]
I think this is too old and also I would just refactor the hole document locking. 
If we don't want to have this PR as reminder we should close it.

**Comment by LittleHuba** [comment]
> I think this is too old and also I would just refactor the hole document locking.
> If we don't want to have this PR as reminder we should close it.

I think it would be wasted time to change something so deeply integrated for this version. We do not have this issue for the rewrite and would be unable to reuse the code anyway. Better to use that time for fixing real bugs.

**Pull request closed without merging**